__tests__/app.test.js

```diff
@@ -68,11 +65,136 @@ describe("PATCH /api/articles/:article_id", () => {
+  test("GET respond with an array of users", async () => {
+    const res = await request(app).get("/api/users").expect(200);
+
+    res.body.user.forEach(() => {
+      expect.objectContaining({
+        username: expect.any(String),
+        name: expect.any(String),
+        avatar_url: expect.any(String),
+      });
+    });
+    expect(res.body.user.length).toBe(4);
+  });
+});
+
+describe("GET /api/articles", () => {
+  test("GET respond with an array of articles with the comment_count", async () => {
+    const res = await request(app).get("/api/articles").expect(200);
+
+    res.body.article.forEach((articles) => {
+      expect(articles).toEqual({
+        article_id: expect.any(Number),
+        title: expect.any(String),
+        topic: expect.any(String),
+        author: expect.any(String),
+        body: expect.any(String),
+        created_at: expect.any(String),
+        votes: expect.any(Number),
+        comment_count: expect.any(Number),
+      });
+    });
+  });
+});
+
+describe("GET /api/articles/article_id/comments", () => {
+  test("GET respond with an array of article comments", async () => {
+    const res = await request(app).get("/api/articles/1/comments").expect(200);
+    expect(res.body.comments.length).toBe(11);
+    res.body.comments.forEach((comments) => {
+      expect(comments).toEqual({
+        comment_id: expect.any(Number),
+        body: expect.any(String),
+        votes: expect.any(Number),
+        author: expect.any(String),
+        article_id: 1,
+        created_at: expect.any(String),
+      });
+    });
+  });
+  test("responds with a 404 status and msg of not found", async () => {
+    const res = await request(app)
+      .get("/api/articles/999/comments")
+      .expect(404);
+    expect(res.body.msg).toBe("not found");
+  });
+});
+
+describe("POST /api/articles/3/comments", () => {
+  test("post a new comment under a username?", async () => {
+    const requestBody = {
+      body: "this is a comment",
+      username: "icellusedkars",
+    };
+    const res = await request(app)
+      .post("/api/articles/1/comments")
+      .send(requestBody)
+      .expect(201);
+    expect(res.body.comment).toEqual({
+      author: "icellusedkars",
+      body: "this is a comment",
+      votes: 0,
+      article_id: 1,
+      created_at: expect.any(String),
+      comment_id: expect.any(Number),
+    });
+  });
+  test("POST return 400 when body doesnt contain all required keys", async () => {
+    const requestBody = {
+      body: "this is a comment",
+    };
+    const res = await request(app)
+      .post("/api/articles/1/comments")
+      .send(requestBody)
+      .expect(400);
+    expect(res.body.msg).toEqual("invalid request");
+  });
+});
+
+describe("GET /api/articles (queries)", () => {
+  test("GET respond with an array of articles sorted by descending order", async () => {
+    const res = await request(app)
+      .get("/api/articles?order=created_at")
+      .expect(200);
```

Comment:
this query should be `sort_by` - the `order` query should be in reference to whether it's `asc`ending or `desc`ending